Ingest dev guide (#1415)
stellar/stellar-docs: docs/data/indexers/build-your-own/ingest-sdk/developer_guide/ledgerbackends/captivecore.mdx

---
title: Captive Core
sidebar_position: 0
---

Captive Core invokes the `stellar-core` binary as a subprocess to stream ledgers from the Stellar network. It can be used to stream a ledger range from the past or to stream new ledgers whenever they are confirmed by the network.

## Prerequisites

> Using captive Core requires stellar-core binary to be installed. Since stellar-core is resource-intensive, it has specific hardware requirements outlined here: [Hardware Requirements](/docs/validators/admin-guide/prerequisites#hardware-requirements).

Reviewer: ```suggestion
Using captive Core requires stellar-core binary to be [installed](/docs/validators/admin-guide/installation) first. 
```
This would avoid repeating details of core aspects. And consider deleting the `Installation` section which follows here, lines 12 - 23, as the core admin guide referral addresses this aspect also.

### Installation

1. Install Stellar Core:

   - Option 1: Build from source by following the [Installation Guide](/docs/validators/admin-guide/installation#installing-from-source)
   - Option 2: Install via a package manager by referring to the [Package-based Installation Guide](/docs/validators/admin-guide/installation#package-based-installation)

2. Verify installation:
   ```bash
   ./stellar-core version
   ```

## Configuration and Usage

> After installing the `stellar-core` binary, you'll need to provide it configuration in TOML format. This configuration requires at a minimum:

Reviewer: ```suggestion
Set the captive core configuration for target Stellar network in TOML format. This configuration requires at a minimum:
```
